crates/polars-python/src/interop/numpy/to_numpy_series.rs

```diff
@@ -340,26 +340,14 @@ where
     let values = ca.iter().map(|v| v.unwrap_or(i64::MIN).into());
     PyArray1::<T>::from_iter_bound(py, values).into_py(py)
 }
-/// Convert lists by flattening first, converting the flat Series, and then splitting by offsets.
 fn list_series_to_numpy(py: Python, s: &Series, writable: bool) -> PyObject {
     let ca = s.list().unwrap();
-    let s_inner = ca.get_inner();
-
-    let np_array_flat = series_to_numpy(py, &s_inner, writable, true).unwrap();
 
-    // Split the NumPy array into subarrays by offset.
```

Comment:
This is wrong for 2 reasons. The slicing was incorrect, but it also doesn't include null values and can show masked out values.